pkg/provider/nomad/nomad.go

```diff
@@ -183,26 +190,52 @@ func (p *Provider) Provide(configurationChan chan<- dynamic.Message, pool *safe.
 			ctx, cancel := context.WithCancel(ctxLog)
 			defer cancel()
 
-			// load initial configuration
-			if err := p.loadConfiguration(ctx, configurationChan); err != nil {
-				return fmt.Errorf("failed to load initial nomad services: %w", err)
+			serviceEventsChan, err := p.pollOrWatch(ctx)
+			if err != nil {
+				logger.Error().Err(err).Msg("Error watching nomad events")
+				return err
```

Comment:
```suggestion
				return errors.New("error watching nomad events")
```

Reply:
Not logging the original error means we won't have any context to know why the watching failed. The pollOrWatch call is the first time the code attempts to connect to Nomad so it would be important to log enough context for diagnoses. We could wrap the error instead?

Reply:
Yes my bad, we have to wrap the error